File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        c=0
        h={0:1}
        cs=0
        for i in nums:
            cs=cs+i
            if cs-k in h:
                c=c+h[cs-k]
            h[cs]=h.get(cs,0)+1
        return c

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        HashMap<Integer,Integer> y=new HashMap<>();
        int res=0;
        int sum=0;
        y.put(0,1);
        for(int j=0;j<nums.length;j++){
            sum+=nums[j];
            int ques=sum-k;
            if (y.containsKey(ques)){
                res+=y.get(ques);
            }
            y.put(sum,y.getOrDefault(sum,0)+1);
        }return res;
        
    }
}